### ISSUE TITLE
Libation crashed while downloading library (8.5.0 Chardonnay Windows)

### ISSUE THREAD

**Issue by got2bQWERTY** [rmcrackan/Libation#389]
**Describe the bug**
First time running Libation 8.5.0 Chardonnay Windows, loaded my library. Tried to leave downloading all titles overnight. When I came back in the morning Libation had crashed after 167 titles (out of 2547 titles). [Side bug, potentially, when processing the library it says 1611 titles processed but Libation only shows 2547 on the bottom of the window. Not sure if there's a reason for this or if some titles are missing]

**To Reproduce**

**Expected behavior**
Expected behavior was for all titles to download and for Libation to remain open.

**Screenshots**
![Screenshot_20221101_104316](https://user-images.githubusercontent.com/117424061/199887836-85df9847-b408-4736-af45-3c4c536b208f.png)

**Platform**

Windows 11 Pro  Version 10.0.22621 Build 22621

**Log Files**
[_Log202211.log](https://github.com/rmcrackan/Libation/files/9935026/_Log202211.log)

**Comment by rmcrackan** [comment]
Thanks for the log. How strange that it crashed without logging an error. Is that D drive out of space? (I know that's a long-shot guess; I want to eliminate weird stuff like that as a possibility.) Is there a chance that the D drive lost connectivity sometime in the night? If you manually start the liberate/download again, what happens? Specifically, does it start downloading again? does it crash right away? If it does crash, does it show an error window or just vanish without a trace?

**Comment by got2bQWERTY** [comment]
The D drive is an external drive, but it has lots of space remaining (4.91TB). I don't think it became disconnected at any point. I have tried restarting the downloads, it starts where it left off (at least I'm assuming it's where it left off). The next two times open it downloaded about 100 titles. The next time after that about 1,000. Each time it crashes it does so without displaying a notification, it just closes. Now, it appears to be stuck on a specific title and gets to 100% but doesn't continue. I've included some screenshots of the progress after each reboot, where it's stuck now, and the new log files (in the Dropbox link below since the log files exceed GitHub's file size limit). If there's anything else I can do to help just let me know.

https://www.dropbox.com/sh/yrw6fqkrudzw0ap/AAABpj5E_ngyZ_XBNAToMEgTa?dl=0

![Screenshot_20221103_114501](https://user-images.githubusercontent.com/117424061/200101370-6f34ffc6-41ed-4f3a-84bb-0c15ccbd4c79.png)
![Screenshot_20221104_080057](https://user-images.githubusercontent.com/117424061/200101380-eb6aba28-97ba-4297-bdb2-4336e28f5a4c.png)
![Screenshot_20221104_103405](https://user-images.githubusercontent.com/117424061/200101394-871b6eb7-cf13-485e-9da4-c7764e07dd00.png)

**Comment by rmcrackan** [comment]
Hmm, Could you rename your log and try again. Libation will create a new log which will be smaller. Hopefully it will show us what's going wrong with this book.

**Comment by got2bQWERTY** [comment]
I'll rename the log and rerun things so you can get a better sense of what's crashing things. Since last night however the stuck title did end up finally downloading. It goes through 194 titles where it cannot download (I believe these are all former Audible Plus catalogue titles which have been removed from the Plus catalogue) then it hangs on whichever book is next. I tried removing the book before it got to that point to see if it would still hang on a different title and same thing, remained stuck at 100%. Approximately an hour later the book seems to have resolved and it moves on to the next title. It completed approximately 200 titles then crashed again. I also tried just doing one title and it got stuck on 100% too.

I ran it a few times throughout the night and it now seems to have finished going through all my library. Libation shows 372 titles "No progress" but scrolling through the list these all appear to be either podcasts or former Plus catalogue titles.

![Screenshot 2022-11-05 013841](https://user-images.githubusercontent.com/117424061/200142926-c52baabf-8bf3-48a3-ab93-09973af03841.png)
![Screenshot 2022-11-05 110759](https://user-images.githubusercontent.com/117424061/200143023-6844b706-c979-47bd-8dba-1b173890e1e3.png)
![Screenshot 2022-11-05 155138](https://user-images.githubusercontent.com/117424061/200143041-53104815-5b08-4633-9172-9c1fcd01db0e.png)

**Comment by got2bQWERTY** [comment]
Just ran it again. Thought everything that was going to download was downloaded, but 178 more items downloaded. Here's the log you requested.

![Screenshot 2022-11-05 182820](https://user-images.githubusercontent.com/117424061/200146783-4e558ffd-ae85-42e1-bd0a-00cf50b6fcba.png)
[Log202211.log](https://github.com/rmcrackan/Libation/files/9944871/Log202211.log)

**Comment by rmcrackan** [comment]
None of the logged errors are crashing the program. Libation runs for another 1/2 hour after the final logged error, so if this ended in a crash, it was from something that didn't get logged. This is a tough one.

Libation has a 'remove book' feature (import \> remove library books) which compares your local list to your audible library and you can select which ones to remove. It will auto-check for removal all books which are no longer in your audible library. This might help you. This removes entries from Libation, it doesn't delete any files.

**Comment by got2bQWERTY** [comment]
Weird that the crashes aren't showing up in the logs.

Regarding removing the old books, this is likely more complicated than scanning the library and is an issue with Audible's library not your program. I don't know if it's the same everywhere, but in Canada we have a Plus catalogue which is a Netflix-style rotating library of books for anybody with a membership. You add them to your library, but they stay in your library even if they are removed from the Plus catalogue. You just can't play them anybody. So Audible still shows these books are "in your library". You can manually remove the books from your library in Audible, but I don't know how to automatically list titles which aren't playable so it would likely take lots of time and guessing. I don't know why Audible does it this way but it is annoying and will probably be hard to adjust for. If you can think of a way to filter out unplayable titles in Libation that would be great, but honestly I don't think this should be a priority for you, especially if this only applies regionally.

![Screenshot 2022-11-06 115811](https://user-images.githubusercontent.com/117424061/200189912-dedf8f1d-f983-4b30-b1bd-1f2648c22ade.png)
![Screenshot 2022-11-06 115833](https://user-images.githubusercontent.com/117424061/200189913-a4e5e45a-691f-4fa6-9714-4c5328b0b0d6.png)

**Comment by rmcrackan** [comment]
> they stay in your library even if they are removed from the Plus catalogue. You just can't play them

Wow. Thanks audible. What a useful feature. My own library of shtuff you won't let me listen to. /s

**Comment by got2bQWERTY** [comment]
> > they stay in your library even if they are removed from the Plus catalogue. You just can't play them
> 
> Wow. Thanks audible. What a useful feature. My own library of shtuff you won't let me listen to. /s

I'd like to believe it was a well-intentioned design choice so people wouldn't get upset at missing titles (even though people should be plenty used to rotating libraries by now). This pop-up that displayed when I tried accessing a title shows why they really did it though...
![Screenshot_20221105-183306](https://user-images.githubusercontent.com/117424061/200228061-53ed7ccf-a55f-49fb-b7a9-0b871cd15d2e.png)

**Comment by rmcrackan** [comment]
Are these weird books the only ones still giving you trouble?

**Comment by got2bQWERTY** [comment]
There are a few podcasts too, but that's not a big deal for me. I get why the Plus books weren't downloading, but don't understand why podcasts randomly weren't. Many would fail one pass then download the next.

Thank you.

**Comment by got2bQWERTY** [comment]
I just had Libation crash and generate an error message. Maybe the log now would help you better understand what was going on.

The log file is 29MB though so I need to do it via Dropbox again, sorry. https://www.dropbox.com/s/79f2u1t9726jiil/_Log202211.log?dl=0

Here's what I did today to achieve the error.
1-Downloaded everything with split chapters. Wanted to redownload as one file. Reset the download status of my entire library, changed the download settings, and tried again in 8.5.0 Chardonnay. Crashed again with the same issue.
2-Wanted to see if a different version of Libation would still encounter the same issues. Ran 8.4.6 Classic. Immediately encountered a number of error messages which kept popping up even if I closed the window. Had to close Libation in task manager (attached a screenshot of the error below).
3-Renamed the existing Libation file and ran 8.4.6 Classic again. No error messages.
4-Reconnected Audible. Input settings. Began downloading again. Seemed to be working fine and going smoother. Everything was happening much quicker. Got through roughly 500 files then popped up an error message. Upon confirming the error message Libation closed. (Please note, the screenshot of today's error message is the one saying "Libation has crashed due to an unhandled error")

Hope this helps you in somehow tracking down what's making it crash. (Sorry, not trying to be a pain in the ass)
![Screenshot 2022-11-07 231855](https://user-images.githubusercontent.com/117424061/200485322-bf4cfb48-fb89-4d15-a852-cea4a0a21960.png)
![Screenshot 2022-11-06 223532](https://user-images.githubusercontent.com/117424061/200485477-95f8cc22-4c1c-48b6-aaaa-c2f73ca4e2cb.png)
![Screenshot 2022-11-06 223545](https://user-images.githubusercontent.com/117424061/200485478-e12092d0-7503-4562-a6a7-798d48ee4a8d.png)
![Screenshot 2022-11-06 224902](https://user-images.githubusercontent.com/117424061/200485482-26967e0a-a108-46d3-853e-ea31f48ebe4d.png)

**Comment by got2bQWERTY** [comment]
[Log202211.log](https://github.com/rmcrackan/Libation/files/9958433/Log202211.log)

**Comment by rmcrackan** [comment]
w00h00 -- Thanks! Finally we something caught something.

One thing is that you have a corrupt file. This has been known to happen but I thought I handled this error correctly -- obviously not. I just wrote the fix. I'm building new versions now. I'll reply here again when it's ready. Then we can see if the new version fixes your issue. fyi -- it will help Chardonnay and Classic. Which reminds me -- thank you so much for being willing to try both. Nice sleuthing :)
